include/bluetooth/direction.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_DF_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_DF_H_
+
+/** @brief Constant Tone Extension parameters for connectionless
+ * transmission.
+ *
+ * The structure holds information required to setup CTE transmission
+ * in periodic advertising.
+ */
+struct bt_df_adv_cte_tx_param {
+	/** Length of CTE in 8us units */
+	uint8_t  cte_len;
+	/** CTE Type: AoA, AoD 1us slots, AoD 2us slots */
+	uint8_t  cte_type;
+	/** Number of CTE to transmit in each periodic adv interval */
+	uint8_t  cte_count;
+	/** Number of Antenna IDs in the switch pattern */
+	uint8_t  num_ant_ids;
+	/** List of antenna IDs in the pattern */
+	uint8_t  *ant_ids;
+};
+
+/** @brief Set or update the Constant Tone Extension parameters for periodic
+ * advertising set.
+ *
+ * Update of Constant Tone Extension parameters is not allowed while
+ * transmission of Constant Tone Extension is enabled on particular extended
```

Comment:
Is the rule about enabling CTE transmission also part of the follow up PR?
I suggest we move this text block until we have added that API so that we can review it in its complete form.

Reply:
The rule is actually related with Core spec: Vol 4, Part E 7.8.80 LE Set Connectionless CTE Transmit Parameters command:
`If the Host issues this command when Constant Tone Extensions have been
enabled in the advertising set, the Controller shall return the error code
Command Disallowed (0x0C).`

Reply:
Right, but need to explain what that means in terms of the zephyr Host API.
When is CTE enabled in the advertising set? periodic_adv start with CTE configured?

Reply:
CTE is enabled by call to another function, the one that will be provided in following PR. 
The function here is responsible for setting CTE parameters and related with HCI command `BT_HCI_OP_LE_SET_CL_CTE_TX_PARAMS`.
The CTE enable is another API call that will be named: `bt_df_set_adv_cte_tx_enabled`. This function is related with `BT_HCI_OP_LE_SET_CL_CTE_TX_ENABLE` HCI command.